x/upgrade/keeper/keeper.go

```diff
@@ -58,6 +59,11 @@ func (k Keeper) SetUpgradeHandler(name string, upgradeHandler types.UpgradeHandl
 	k.upgradeHandlers[name] = upgradeHandler
 }
 
+// DeleteUpgradeHandler removes an UpgradeHandler for the given upgrade name
+func (k Keeper) DeleteUpgradeHandler(name string) {
```

Comment:
It's used only in tests, can we avoid test speicific functions here? Not a blocker but this smells bad

Reply:
Agreed, would be good to not publicly expose this.

Reply:
> It's used only in tests, can we avoid test speicific functions here? 


@anilCSE I don't find another way to add testcase for `no upgrade handler is present for last upgrade`. Shall we remove that testcase?

Reply:
Yes, let's remove it and have a follow up issue..may be add a TODO and remove this